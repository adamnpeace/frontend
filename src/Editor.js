import React from "react";
import PropTypes from "prop-types";
import AceEditor from "react-ace";
import "brace/theme/textmate";
import { Button } from "grommet";

class Editor extends React.Component {
  constructor(props) {
    super(props);
  }
  static propTypes = {
    computeOutput: PropTypes.func.isRequired
  };

  render() {
    let input;
    const style = {
      fontSize: "14px !important",
      border: "1px solid lightgray",
      width: "100%",
      height: "100%"
    };

    const { computeOutput } = this.props.computeOutput;
    return (
      <form
        onSubmit={e => {
          e.preventDefault();
          computeOutput(input.value);
          input.value = "";
        }}
      >
        <AceEditor
          style={style}
          mode="python"
          theme="textmate"
          name="blah2"
          onLoad={this.onLoad}
          onChange={this.onChange}
          fontSize={14}
          showPrintMargin={true}
          showGutter={true}
          highlightActiveLine={true}
          value={this.props.default}
          setOptions={{
            enableBasicAutocompletion: false,
            enableLiveAutocompletion: false,
            enableSnippets: false,
            showLineNumbers: true,
            tabSize: 2
          }}
        />
        <input
          className="form-control col-md-12"
          ref={node => {
            input = node;
          }}
        />
        <br />
        <Button label="Submit" />
      </form>
    );
  }
}
export default Editor;
