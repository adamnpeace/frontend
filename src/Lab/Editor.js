import React from "react";
import AceEditor from "react-ace";
import "brace/theme/textmate";
import "brace/mode/python";
import axios from "axios";
import {Config} from "../Config";
import * as Icons from "grommet-icons";
import {Box, Button, Heading, Text} from "grommet";

class Editor extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      value: this.props.code,
      snippet: "",
      result: ""
    };
    this.onChange = this.onChange.bind(this);
  }
  
  componentDidMount() {
    this.setState({value: this.props.code});
  }
  
  componentDidUpdate(prevProps, prevState) {
    if (this.state.snippet !== prevState.snippet) {
      axios
        .get(Config.apiUrl + "test/", {
          params: {model_input: this.state.snippet}
        })
        .then(res => {
          this.setState({result: res.data.complex_result});
        });
    }
  }
  
  computeOutput(snippet) {
    this.setState({snippet: snippet});
  }
  
  onChange(newValue) {
    this.setState({
      value: newValue
    });
  }
  
  render() {
    const style = {
      fontSize: "14px !important",
      border: "1px solid lightgray",
      width: "100%",
      height: "100%"
    };
    return this.state.result !== "" ? (
      <Box column gap="small" height="80vh">
        <Box height="70%">
          <AceEditor
            style={style}
            mode="python"
            theme="textmate"
            name="blah2"
            onLoad={this.onLoad}
            onChange={this.onChange}
            fontSize={14}
            showPrintMargin={true}
            showGutter={true}
            highlightActiveLine={true}
            value={this.state.value}
            setOptions={{
              enableBasicAutocompletion: false,
              enableLiveAutocompletion: false,
              enableSnippets: false,
              showLineNumbers: true,
              tabSize: 2
            }}
          />
        </Box>
        <Box height="30%" overflow="auto">
          <Button
            color="accent-1"
            primary
            icon={<Icons.Edit/>}
            label="Submit"
            onClick={e => {
              this.computeOutput(this.state.value);
            }}
          />
          <Heading level={3}>Output:</Heading>
          <Text>{this.state.result}</Text>
        </Box>
      </Box>
    ) : (
      <Box column gap="small" height="80vh">
        <Box height="100%" elevation="xsmall">
          <AceEditor
            style={style}
            mode="python"
            theme="textmate"
            name="blah2"
            onLoad={this.onLoad}
            onChange={this.onChange}
            fontSize={14}
            showPrintMargin={true}
            showGutter={true}
            highlightActiveLine={true}
            value={this.state.value}
            setOptions={{
              enableBasicAutocompletion: false,
              enableLiveAutocompletion: false,
              enableSnippets: false,
              showLineNumbers: true,
              tabSize: 2
            }}
          />
        </Box>
        <Box>
          <Button
            color="accent-1"
            primary
            icon={<Icons.Edit/>}
            label="Submit"
            onClick={e => {
              this.computeOutput(this.state.value);
            }}
          />
        </Box>
      </Box>
    );
  }
}

export default Editor;
