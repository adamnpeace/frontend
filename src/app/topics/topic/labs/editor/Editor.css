@media (min-width: 750px) {
  .ace-editor {
    top: 3.25rem;
    height: calc(100vh - 6.5rem);
    width: 50vw;
  }
}

@media (max-width: 750px) {
  .ace-editor {
    top: 3.25rem;
    height: calc(50vh - 3.25rem);
    width: 100vw;
  }
}
