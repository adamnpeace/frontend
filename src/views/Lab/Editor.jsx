import React from 'react';
import AceEditor from 'react-ace';
import { Prompt } from 'react-router-dom';
import 'brace/theme/textmate';
import 'brace/mode/python';
import axios from 'axios';
import * as Icons from 'grommet-icons';
import { Box, Button, Heading, Text } from 'grommet';
import PropTypes from 'prop-types';
import { Config } from '../../Config';

const propTypes = {
  codeSnippet: PropTypes.string.isRequired,
  changeCode: PropTypes.func.isRequired
};

const defaultProps = {};

class Editor extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      isBlocking: false,
      codeSnippet: props.codeSnippet,
      result: ''
    };
    this.onChangeCode = this.onChangeCode.bind(this);
  }

  onChangeCode(newValue) {
    this.setState({
      isBlocking: newValue.length > 0,
      codeSnippet: newValue
    });
    this.props.changeCode(this.state.codeSnippet);
  }

  computeOutput(snippet) {
    axios
      .get(`${Config.apiUrl}test/`, {
        params: { model_input: snippet }
      })
      .then(res => {
        this.setState({ result: res.data.complex_result });
      });
  }

  render() {
    const style = {
      fontSize: '14px !important',
      border: '1px solid lightgray',
      width: '100%',
      height: '100%'
    };

    const { isBlocking, codeSnippet, result } = this.state;

    return this.state.result !== '' ? (
      <Box column gap="small" height="85vh">
        <Prompt
          when={isBlocking}
          message={() => `You're editing, are you sure you want to leave this page?`}
        />
        <Box height="70%">
          <AceEditor
            style={style}
            mode="python"
            theme="textmate"
            name="blah2"
            onLoad={this.onLoad}
            onChange={this.onChangeCode}
            fontSize={14}
            showPrintMargin
            showGutter
            highlightActiveLine
            value={codeSnippet}
            setOptions={{
              enableBasicAutocompletion: false,
              enableLiveAutocompletion: false,
              enableSnippets: false,
              showLineNumbers: true,
              tabSize: 2
            }}
          />
        </Box>
        <Box height="30%" overflow="auto">
          <Button
            color="accent-1"
            primary
            icon={<Icons.Edit />}
            label="Submit"
            onClick={e => {
              this.computeOutput(codeSnippet);
            }}
          />
          <Heading level={3}>Output:</Heading>
          <Text>{result}</Text>
        </Box>
      </Box>
    ) : (
      <Box column gap="small" height="80vh">
        <Prompt
          when={isBlocking}
          message={() => `You're editing, are you sure you want to leave this page?`}
        />
        <Box height="100%" elevation="xsmall">
          <AceEditor
            style={style}
            mode="python"
            theme="textmate"
            name="blah2"
            onLoad={this.onLoad}
            onChange={this.onChangeCode}
            fontSize={14}
            showPrintMargin
            showGutter
            highlightActiveLine
            value={codeSnippet}
            setOptions={{
              enableBasicAutocompletion: false,
              enableLiveAutocompletion: false,
              enableSnippets: false,
              showLineNumbers: true,
              tabSize: 2
            }}
          />
        </Box>
        <Box>
          <Button
            color="accent-1"
            primary
            icon={<Icons.Edit />}
            label="Submit"
            onClick={e => {
              this.computeOutput(codeSnippet);
            }}
          />
        </Box>
      </Box>
    );
  }
}

Editor.propTypes = propTypes;
Editor.defaultProps = defaultProps;

export default Editor;
