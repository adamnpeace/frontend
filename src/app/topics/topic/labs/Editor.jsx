import React, { useState, useEffect } from 'react';
import AceEditor from 'react-ace';
import axios from 'axios';
import 'brace/mode/python';
import 'brace/snippets/python';
import 'brace/theme/xcode';
import { withRouter } from 'react-router-dom';
import { useGlobalState } from '../../../../state';
import './editor/Editor.css';
import { apiUrl } from '../../../../config';

Editor.defaultProps = {
  lessonNum: 0,
  codeSnippet: `class Hei:
  def hello(var):
    output = "Hi"
    print(output)`
};

function Editor(props) {
  const { lessonNum, codeSnippet } = props;
  console.log(lessonNum);
  const [lastLessonNum, setLessonNum] = useState(0);
  const [currentCode, setCode] = useState(codeSnippet);

  useEffect(() => {
    if (lessonNum !== lastLessonNum) {
      setLessonNum(lessonNum);
      setCode(codeSnippet);
    }
  });

  function onChange(currentCode) {
    setCode(currentCode);
  }

  function computeOutput(snippet) {
    axios
      .get(`${apiUrl}test/`, {
        params: { model_input: snippet }
      })
      .then(res => {
        this.setState({ result: res.data.complex_result });
      });
  }

  const style = {
    'min-height': '40vh',
    height: '100%',
    width: '100%'
  };

  return (
    <div className="ace-editor">
      <AceEditor
        style={style}
        mode="python"
        theme="xcode"
        name="blah2"
        /*onLoad={this.onLoad}*/
        onChange={onChange}
        fontSize={16}
        showPrintMargin={true}
        showGutter={true}
        highlightActiveLine={false}
        value={currentCode}
        setOptions={{
          enableBasicAutocompletion: true,
          enableLiveAutocompletion: true,
          enableSnippets: true,
          showLineNumbers: true,
          tabSize: 2
        }}
      />
    </div>
  );
}
export default Editor;
