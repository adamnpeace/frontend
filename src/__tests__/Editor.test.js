import React from "react";
import ReactDOM from "react-dom";
import Editor from '../LabView/Editor';
import renderer from "react-test-renderer";

describe("Editor Component Test", () => {
  it("Editor renders without crashing", () => {
    const div = document.createElement('div');
    ReactDOM.render(<Editor />, div);
  });

  it("Editor renders correctly according to first snapshot", () => {
    const tree = renderer.create(<Editor />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
