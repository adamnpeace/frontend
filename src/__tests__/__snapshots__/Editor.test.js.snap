// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Editor Component Test Editor renders correctly according to first snapshot 1`] = `
<div
  className="StyledBox-sc-13pk1d4-0 cscWAI"
>
  <div
    className="StyledBox-sc-13pk1d4-0 gpqUzJ"
  >
    <div
      id="blah2"
      style={
        Object {
          "border": "1px solid lightgray",
          "fontSize": "14px !important",
          "height": "100%",
          "width": "100%",
        }
      }
    />
  </div>
  <div
    className="StyledBox__StyledBoxGap-sc-13pk1d4-1 hPsznL"
  />
  <div
    className="StyledBox-sc-13pk1d4-0 IRSNj"
  >
    <button
      className="StyledButton-sc-323bzc-0 jWMJbT"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      type="button"
    >
      <div
        className="StyledBox-sc-13pk1d4-0 hHwlHq"
      >
        <svg
          aria-label="Edit"
          className="StyledIcon-ofa7kd-0 cZXyLU"
          viewBox="0 0 24 24"
        >
          <path
            d="M14,4 L20,10 L14,4 Z M22.2942268,5.29422684 C22.6840146,5.68401459 22.6812861,6.3187139 22.2864907,6.71350932 L9,20 L2,22 L4,15 L17.2864907,1.71350932 C17.680551,1.319449 18.3127724,1.31277239 18.7057732,1.70577316 L22.2942268,5.29422684 Z M3,19 L5,21 M7,17 L15,9"
            fill="none"
            stroke="#000"
            strokeWidth="2"
          />
        </svg>
        <div
          className="StyledBox__StyledBoxGap-sc-13pk1d4-1 iChEkS"
        />
        Submit
      </div>
    </button>
  </div>
</div>
`;
